tobac/feature_detection.py

```diff
@@ -16,35 +16,44 @@
    diverse datasets. Geoscientific Model Development,
    12(11), 4551-4570.
 """
-
+from __future__ import annotations
+from typing import Union
+import warnings
 import logging
+
+from typing_extensions import Literal
```

Comment:
I think we should consider moving the requirements to python 3.8 for the 1.6 release, which would remove this dependency

Reply:
I guess the question is do we want to get this in 1.5.2 or wait for 1.6.0. I'm inclined to push it through and use the typing extensions, but I'm not committed to that plan.

Reply:
I'm happy to use `typing_extensions` for now and make a note to change it for 1.6

Reply:
Agree with the above